src/lib/core/style/focus-classes.spec.ts

```diff
@@ -32,21 +22,27 @@ describe('FocusOriginMonitor', () => {
     TestBed.compileComponents();
   }));
 
-  beforeEach(inject([FocusOriginMonitor, Platform], (fom: FocusOriginMonitor, pfm: Platform) => {
+  beforeEach(inject([FocusOriginMonitor], (fom: FocusOriginMonitor) => {
     fixture = TestBed.createComponent(PlainButton);
     fixture.detectChanges();
 
     buttonElement = fixture.debugElement.query(By.css('button')).nativeElement;
     buttonRenderer = fixture.componentInstance.renderer;
     focusOriginMonitor = fom;
-    platform = pfm;
 
     focusOriginMonitor.registerElementForFocusClasses(buttonElement, buttonRenderer);
+
+    // On Saucelabs, browsers will run simultaneously and therefore can't focus all browser windows
+    // at the same time. This is problematic when testing focus states. Chrome and Firefox
+    // only fire FocusEvents when the window is focused. This issue also appears locally.
+    let _nativeButtonFocus = buttonElement.focus.bind(buttonElement);
```

Comment:
can you extract this into a separate method e.g. `fixFocus(el: Element)`? also both this and `dispatchFocusEvent` would be good things to put in a separate test utilities directory